backend/tests/Tests/Execution.Tests.fs

```diff
@@ -600,6 +616,172 @@ let testMatchPreview : Test =
       // TODO: constructor around a constructor around a value
       ]
 
+
+let testLetPreview : Test =
+  let letID = gid ()
+
+  let createLetPattern
+    (patternMatch : LetPattern)
+    (expr : Expr)
+    (finalExpr : Expr)
+    : Task<Map<id, AT.ExecutionResult>> =
+    task {
+      let ast = ELet(letID, patternMatch, expr, finalExpr)
+
+      let! results = execSaveDvals "let-preview" [] [] [] ast
+      let stuff = results |> Dictionary.toList |> Map
+
+      return stuff
+    }
+
+  testList
+    "test let preview"
+    [ testTask "let hello = 1 in hello" {
+        let lpID = gid ()
+
+        let letPattern = LPVariable(lpID, "hello")
+        let assignExpr = eInt 1
+        let retExpr = eVar "hello"
+
+        let! result = createLetPattern letPattern assignExpr retExpr
+
+        Expect.equal
+          (Map.get lpID result)
+          (Some(AT.ExecutedResult(DInt 1L)))
+          "hello assigned correctly"
+      }
+
+      testTask "let (x, y) = (1, 2) in y" {
+        let lpID = gid ()
+        let xID = gid ()
+        let yID = gid ()
+
+        let letPattern =
+          LPTuple(lpID, LPVariable(xID, "x"), LPVariable(yID, "y"), [])
+        let assignExpr = eTuple (eInt 1) (eInt 2) []
+        let retExpr = eVar "y"
+
+        let! result = createLetPattern letPattern assignExpr retExpr
+
+        Expect.equal
+          (Map.get xID result)
+          (Some(AT.ExecutedResult(DInt 1L)))
+          "x assigned correctly"
+
+        Expect.equal
+          (Map.get yID result)
+          (Some(AT.ExecutedResult(DInt 2L)))
+          "y assigned correctly"
+      }
+
+      testTask "let (x, y) = (1, 1/0) in y" {
+        let lpID = gid ()
+        let xID = gid ()
+        let yID = gid ()
+        let divID = gid ()
+
+        let letPattern =
+          LPTuple(lpID, LPVariable(xID, "x"), LPVariable(yID, "y"), [])
+
+        // Should result in type error
+        let divisionExpr =
+          EApply(
+            divID,
+            PT.FQFnName.stdlibFqName "Int" "divide" 0
+            |> PT2RT.FQFnName.toRT
+            |> FnName,
+            [],
+            [ eInt 1; eInt 0 ]
+          )
+
+        let assignExpr = eTuple (eInt 1) divisionExpr []
+        let retExpr = eVar "y"
+
+        let! result = createLetPattern letPattern assignExpr retExpr
+
+        // TODO: remove
+        debuG "divID" (Map.get divID result) // DEBUG: divID (Some(ExecutedResult (DError (SourceNone, "Division by zero"))))
+        debuG "tuple" (Map.get (Expr.toID assignExpr) result) // DEBUG: tuple (Some(ExecutedResult (DError (SourceNone, "Division by zero"))))
+
+
+        // TODO: figure it out
+        // expected: Some (NonExecutedResult (DInt 1L))
+        // actual: Some (NonExecutedResult (DIncomplete (SourceID (7UL, 309035826UL))))
+        Expect.equal
```

Comment:
https://github.com/darklang/dark/pull/4840/files#r1170419108
@StachuDotNet 
I assume the inner tuple error propagates to the tuple and the tuple also evaluates to a DError.

Does this mean that the implementation has to be changed a little bit?